ext/HealthBaseOMOPCDMExt.jl

```diff
@@ -0,0 +1,108 @@
+module HealthBaseOMOPCDMExt
+
+using HealthBase
+using DataFrames
+using OMOPCommonDataModel
+using Serialization
+using InlineStrings
+using Dates
+
+const OMOPCDM_VERSIONS = Dict{Any, Any}()
+
+# Mapping OMOP CDM datatypes to Julia types
+const datatype_map = Dict(
+    "integer" => Int64, "Integer" => Int64, "bigint" => Int64,
+    "float" => Float64,
+    "date" => Date, "datetime" => DateTime,
+    "varchar(1)" => String, "varchar(2)" => String, "varchar(3)" => String,
+    "varchar(9)" => String, "varchar(10)" => String, "varchar(20)" => String,
+    "varchar(25)" => String, "varchar(50)" => String, "varchar(80)" => String,
+    "varchar(250)" => String, "varchar(255)" => String, "varchar(1000)" => String,
+    "varchar(2000)" => String, "varchar(MAX)" => String
+)
+
+function __init__()
+    @info "OMOP CDM extension for HealthBase has been loaded!"
+    merge!(OMOPCDM_VERSIONS, deserialize(joinpath(@__DIR__, "..", "assets", "version_info")))
+end
+
+"""
+    HealthTable(df::DataFrame; omop_cdm_version="v5.4.0", disable_type_enforcement=false, collect_errors=true)
+
+Constructs a `HealthTable` for an OMOP CDM dataset by validating the given `DataFrame`.
+
+This constructor validates the `DataFrame` against the specified OMOP CDM version. It checks that
+column names are valid OMOP CDM fields and that their data types are compatible. It then
+attaches all available metadata from the OMOP CDM specification to the `DataFrame`'s columns.
+
+By default, it checks all columns and then throws a single `ArgumentError` that lists all
+columns with type mismatches. This behavior can be modified with the keyword arguments below.
+
+## Arguments
+- `df::DataFrame`: The `DataFrame` to wrap. It should contain columns corresponding to an OMOP CDM table.
+
+## Keyword Arguments
+- `omop_cdm_version::String="v5.4.0"`: The OMOP CDM version to validate against.
+- `disable_type_enforcement::Bool=false`: If `true`, type mismatches will emit a single comprehensive warning instead of throwing an error.
+- `collect_errors::Bool=true`: If `false`, the constructor will throw an error immediately upon finding the first column with a type mismatch. If `true` (the default), it will collect all errors and report them in a single message.
+
+## Returns
+- `HealthTable`: A new `HealthTable` instance with validated data and attached metadata.
+
+## Examples
+Examples will be added in a future update as more functionality is integrated.
```

Comment:
I think some examples that would be good to add here are:

1. Loading a DataFrame into a HealthTable (using a DataFrame from scratch with OMOP CDM data)
2. Taking a DataFrame that results from a query into an OMOP CDM database and turning that into a HealthTable
3. How to access metadata about the columns
4. Quick-fail and warning examples for loading data that may not be correctly formatted

This would be perfect; we may put these in actual documentation later, but for now, docstrings are great.